lib/spack/llnl/util/filesystem.py

```diff
@@ -282,6 +284,60 @@ def working_dir(dirname, **kwargs):
 
 
 @contextmanager
+def move_transaction(directory_name, parent=None):
```

Comment:
I think the name `replace_transaction` would be more accurate. I like `replace_directory_transaction` even better.

Reply:
`replace_directory_transaction` is actually a much better fit here.